torch/distributed/_spmd/distribute.py

```diff
@@ -162,32 +162,50 @@ def _remap_arg(arg: object) -> object:
         )
         node_to_obj[node] = out
 
-    op_schema = DTensor._propagator.prepare_op_schema(op_overload, args, kwargs)
-    # get DTensor specs for inputs and outputs
-    output_sharding = DTensor._propagator.propagate_op_sharding(
-        op_overload,
-        op_schema,
-    )
+        op_schema = DTensor._propagator.prepare_op_schema(
+            op_overload, args, kwargs
+        )
+        # get DTensor specs for inputs and outputs
+        output_sharding = DTensor._propagator.propagate_op_sharding(
+            op_overload,
+            op_schema,
+        )
 
-    assert output_sharding.schema_suggestions is not None
-    target_schema = output_sharding.schema_suggestions[0]
-    redistribute = target_schema is not op_schema
+        assert output_sharding.schema_suggestions is not None
+        target_schema = output_sharding.schema_suggestions[0]
+        redistribute = target_schema is not op_schema
 
-    # TODO: this is broken when kwargs contains tensors
-    # or if a non-tensor kwarg was modified by the sharding propagation
-    # (in order to fix, need to port over pack_args_kwargs_with_local_tensor for kwargs as well)
-    updated_args_spec, unflattened_args = _update_specs_for_redistribute(
-        args, target_schema, redistribute
-    )
+        # TODO: this is broken when kwargs contains tensors
+        # or if a non-tensor kwarg was modified by the sharding propagation
+        # (in order to fix, need to port over pack_args_kwargs_with_local_tensor for kwargs as well)
+        updated_args_spec, unflattened_args = _update_specs_for_redistribute(
+            args, target_schema, redistribute
+        )
 
-    dispatch = partial(
-        _dispatch_with_local_tensors,
-        op_overload,
-        kwargs=kwargs,
-        specs=updated_args_spec,
-    )
+        dispatch = partial(
+            _dispatch_with_local_tensors,
+            op_overload,
+            kwargs=kwargs,
+            specs=updated_args_spec,
+        )
 
-    return make_fx(dispatch)(unflattened_args)
+        gm = make_fx(dispatch, _allow_non_fake_inputs=False)(unflattened_args)
```

Comment:
Is it still safe to do DCE? I was informed that with the new tracing, we should avoid using DCE. Or it is safe here because this is before inplace ops? cc., @yifuwang

Reply:
this is DCE within one DTensor-expanded operator subgraph. This is before calling into `Override.transform` and before graph optimization. There are some additional DCE in `distributed.py`. @yifuwang @fegin let me know if I should remove all of those. IIUC, for individual dense op DTensor expansion, this might be OK? Because after @kumpera's `WaitTensor` change, it is now correctly connected to graph output.